app/pages/lab/feedback/feedback-editor.jsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import Translate from 'react-translate-component';
+import counterpart from 'counterpart';
+
+import feedbackTypes from './types';
+
+counterpart.registerTranslations('en', {
+  feedbackEditor: {
+    title: 'Feedback',
+    enable: 'Enable feedback on this task',
+    notAvailable: 'Feedback is not available for this task.',
+    help: 'Feedback can help your volunteers improve their accuracy by checking their classification of known subjects, and reporting back on the results in the classification summary.',
+  }
+});
+
+export default class FeedbackEditor extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleEnableFeedback = this.handleEnableFeedback.bind(this)
+    this.handleFeedbackChange = this.handleFeedbackChange.bind(this)
+    this.isFeedbackAvailable = this.isFeedbackAvailable.bind(this)
+    this.renderFeedbackEnable = this.renderFeedbackEnable.bind(this)
+    this.renderFeedbackForm = this.renderFeedbackForm.bind(this)
+  }
+
+  render() {
+    // Unfortunately, the JSON client mutates an object on update, rather than
+    // returning a new instance. So prevProps will always be identical to current
+    // props, breaking some of the lifecycle methods. This ensures we always have
+    // a new object.
```

Comment:
I don't think it can be helped. It'll be less of a problem once state starts getting moved into Redux, since the reducer can be configured to always return a new object (or more likely, since best practice requires Redux objects to be serialisable, the static data is kept in state and components create a new Resource instance based on that).